### ISSUE TITLE
PsuIdentity User.from_omniauth edge cases

### ISSUE THREAD

**Issue by ajkiessl** [psu-stewardship/researcher-metadata#410]
1. When PSU Identity returns nil (no user found).  I can't think of a legitimate reason we'd get no user. We should only be calling this method when they're logging into the application, and they wouldn't be able to do that without first going through Azure. It's seems impossible for someone to be authenticated via Azure and not be in PSU's identity system. But, that's an assumption. If it were to happen, that seems to be a "WHAAA!!???!!!" kinda moment. Maybe throw something in Bugsnag and return to the root page with "Sorry, we can't log you in"
2. When PSU Identity is down (raises IdentityServiceError).  We can probably defer this to after Fix sign-in so that it is not dependent on identity services #364 is completed, then the service can raise its error, and we do the Bugsnag/redirect to root dance.
3. When PSU Identity returns a successful response, but that response is deemed invalid by the User model (first or last name is blank).  This seems to be more legitimate than the first case, but if we've made our User model incongruent with Penn State's own identity data, that's a shame-on-us moment. Otherwise, it seems unlikely that an Azure-authenticated user wouldn't have both first and last names. Either way, I think the response would the same: "WHAAA??!!??"

**Comment by anaelizabethenriquez** [comment]
@ajkiessl I don't think this is devops, but can you take a look at this issue and see if it should be included in this round of development?

**Comment by ajkiessl** [comment]
@anaelizabethenriquez Correct, this isn't deveops.  #1 and #3 above are for handling extreme edge cases that I'm not too concerned about and would consider low priority.  If those issues do start to pop up, then maybe we would want to address them, but I haven't seen any errors yet that would indicate those edge cases are happening.  #2 is mostly described in the other issue it references (#364).  I still don't consider this very high priority, but it is more likely to happen than the other two edge cases.

I would keep #364 for potential development this round, but low priority, and keep this issue #410 out of current development.
